reactor-core/src/main/java/reactor/util/Loggers.java

```diff
@@ -442,18 +442,45 @@ public void error(String msg, Throwable t) {
 		}
 
 		@Nullable
-		final String format(@Nullable String from, @Nullable Object... arguments){
-			if(from != null) {
+		String format(@Nullable String from, @Nullable Object... arguments) {
+			return format(from, false, arguments);
+		}
+
+		@Nullable
+		private String format(@Nullable String from, boolean skipLast, @Nullable Object... arguments) {
+			if (from != null) {
 				String computed = from;
 				if (arguments != null && arguments.length != 0) {
-					for (Object argument : arguments) {
-						computed = computed.replaceFirst("\\{\\}", Matcher.quoteReplacement(String.valueOf(argument)));
+					int lastIndex = arguments.length;
+					if (skipLast) --lastIndex;
```

Comment:
Please make sure you use braces around the instructions following an if condition.

Reply:
Reflect it!